"""
Copy this snakefile to your raw data directory and modify the LIBRARY to your actual project.
The raw data filename must be "{LIBRARY}.R1.fastq.gz and {LIBRARY}.R2.fastq.gz"
"""
LIBRARY = "H16A28P250-1"
"""
run the shell:
export PATH=/home/xujm/soft/Anaconda/anaconda3/bin:$PATH
source activate snakemake
snakemake -j 80 --cluster-config cluster.json -c "qsub -cwd -l vf={cluster.vf} -e {cluster.err} -o {cluster.std}" --jn {params.job_name}{jobid} -T
"""
import os
SPLIT_FILES_NUM = 50
"""
Primer setting:
There are two built-in primers: hiseq and miseq, or you can set primer personality
"""
PRIMER_NAME = "hiseq"
F_PRIMER = ""
R_PRIMER = ""
"""
sam_barcode.all is space blank separated file, each line contains:
1. contract
2. sample
3. barcode(eg: Built-in paired:F1+R1, Single on reverse(12bases): AGATTGACCAAC)
4. data_type(16S,ITS,ARCH)
5. lib_type(hiseq,miseq)
"""
SAM_BARCODE_INFO = "sam_barcode.all"
QUA_MIN = 20

if os.path.exists("logs") != True:
    os.system("mkdir logs")

localrules: all, report

include_prefix = "/data_center_01/home/xujm/bin/.self/Pipeline_dev/16S_QC/"
include:
    include_prefix + "/workflow/config.py"
include:
    include_prefix + "/rules/split_barcode.rules"
include:
    include_prefix + "/rules/pandaseq.rules"
include:
    include_prefix + "/rules/fastqfilter.rules"
include:
    include_prefix + "/rules/report.rules"
include:
    include_prefix + "/rules/fastqc.rules"

rule all:
    input:
        LIBRARY + ".report.html"